contracts/FlowStakingCollection.cdc

```diff
@@ -0,0 +1,844 @@
+                    self.unlockedTokensUsed = self.unlockedTokensUsed - totalStaked
+
+                    /// Removes the NodeDelegator object from the Staking Collections internal nodeDelegators map.
+                    let nodeDelegator <- self.nodeDelegators[nodeID] <- nil
+
+                    return <- nodeDelegator
+                } else { 
+                    panic("Expected delegatorID does not correspond to the delegator in the Staking Collection.")
+                }
+            } else {
+                /// The function does not allow for removing a NodeDelegator stored in the locked account, if one exists.
+                panic("Cannot remove delegator stored in locked account.")
+            }
+
+            return nil
+        }
+
+        /// Operations to register new staking objects
+
+        /// Function to register a new Staking Record to the Staking Collection
+        pub fun registerNode(id: String, role: UInt8, networkingAddress: String, networkingKey: String, stakingKey: String, amount: UFix64) {
+
+            let tokens <- self.getTokens(amount: amount)
+
+            let nodeStaker <- FlowIDTableStaking.addNodeRecord(id: id, role: role, networkingAddress: networkingAddress, networkingKey: networkingKey, stakingKey: stakingKey, tokensCommitted: <-tokens)
+
+            //emit NewNodeCreated(nodeID: nodeStaker.id, role: nodeStaker.id, amountCommitted: amount)
+
+            self.nodeStakers[id] <-! nodeStaker
+        }
+
+        /// Function to register a new Delegator Record to the Staking Collection
+        pub fun registerDelegator(nodeID: String, amount: UFix64) {
+            let delegatorIDs = self.getDelegatorIDs()
+            for idInfo in delegatorIDs {
+                if idInfo.delegatorNodeID == nodeID { panic("Cannot register a delegator for a node that is already being delegated to") }
+            }
+            
+            let tokens <- self.getTokens(amount: amount)
+
+            let nodeDelegator <- FlowIDTableStaking.registerNewDelegator(nodeID: nodeID)
+
+            nodeDelegator.delegateNewTokens(from: <- tokens)
+
+            //emit NewDelegatorCreated(nodeID: nodeDelegator.nodeID, delegatorID: nodeDelegator.id, amountCommitted: amount)
+
+            self.nodeDelegators[nodeDelegator.nodeID] <-! nodeDelegator
+        }
+
+        /// Borrows a reference to a node in the collection
+        access(self) fun borrowNode(_ nodeID: String): &FlowIDTableStaking.NodeStaker? {
+            if self.nodeStakers[nodeID] != nil {
+                return &self.nodeStakers[nodeID] as? &FlowIDTableStaking.NodeStaker
+            } else {
+                return nil
+            }
+        }
+
+        /// Borrows a reference to a delegator in the collection
+        access(self) fun borrowDelegator(_ nodeID: String, _ delegatorID: UInt32): &FlowIDTableStaking.NodeDelegator? {
```

Comment:
I'd recommend using argument labels here, this is exactly what they are for: Here, two IDs are passed, but in a function call arguments get mixed up:
```suggestion
        access(self) fun borrowDelegator(nodeID: String, delegatorID: UInt32): &FlowIDTableStaking.NodeDelegator? {
```

I'd also add them for consistency, `removeDelegator` has the argument labels